Author: Eli <epicadocan@gmail.com>
Description: add support for Linux 3.14
 Patch based on commits created by Krzysztof Kolasa
 from https://github.com/kolasa/fglrx-13.25
Bug-Debian: https://bugs.debian.org/758003

diff -uprN fglrx-8.97.100.7/firegl_public.h fglrx-8.97.100.7-b/firegl_public.h
--- a/common/lib/modules/fglrx/build_mod/firegl_public.h	2013-01-15 22:33:27.000000000 +0100
+++ b/common/lib/modules/fglrx/build_mod/firegl_public.h	2014-08-13 09:27:55.000000000 +0200
@@ -231,8 +231,11 @@ typedef struct {
  */
 typedef int             KCL_TYPE_Pid;       /** Process identifier */
 typedef int             KCL_TYPE_Tgid;      /** Thread Group identifier */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 12, 0)
 typedef int             KCL_TYPE_Uid;       /** User identifier */
-
+#else
+typedef kuid_t KCL_TYPE_Uid; /** User identifier */
+#endif
 /** Atomic variables
  * This type is defined using struct to make sure compiled code will
  * always refer to the memory containing the atomic variable (prevent
diff -uprN fglrx-8.97.100.7/kcl_acpi.c fglrx-8.97.100.7-b/kcl_acpi.c
--- a/common/lib/modules/fglrx/build_mod/kcl_acpi.c	2013-07-23 17:14:36.000000000 +0200
+++ b/common/lib/modules/fglrx/build_mod/kcl_acpi.c	2014-08-13 09:27:55.000000000 +0200
@@ -776,7 +776,11 @@ unsigned int ATI_API_CALL KCL_ACPI_GetHa
 {
 #if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,12)
     #if LINUX_VERSION_CODE >= KERNEL_VERSION(3,8,0)
-        pInfo->video_handle = pInfo->pcidev->dev.acpi_node.handle;
+		#if LINUX_VERSION_CODE < KERNEL_VERSION(3,13,0)
+			pInfo->video_handle = pInfo->pcidev->dev.acpi_node.handle;
+		#else
+			pInfo->video_handle = pInfo->pcidev->dev.acpi_node.companion;
+		#endif
     #elif LINUX_VERSION_CODE > KERNEL_VERSION(2,6,19)
         pInfo->video_handle = pInfo->pcidev->dev.archdata.acpi_handle;
     #else 
